src/pkg/services/m365/api/lists.go

```diff
@@ -529,6 +551,62 @@ func setAdditionalDataByColumnNames(
 	return filteredData
 }
 
+func hasAddressFields(additionalData map[string]any) (map[string]any, string, bool) {
+	for fieldName, value := range additionalData {
+		nestedFields, ok := value.(map[string]any)
+		if !ok || keys.HasKeys(nestedFields, GeoLocFieldName) {
+			continue
+		}
+
+		if keys.HasKeys(nestedFields, addressFieldNames...) {
+			return nestedFields, fieldName, true
+		}
+	}
+
+	return nil, "", false
+}
+
+func concatenateAddressFields(addressFields map[string]any) string {
+	concatenatedAddress := ""
+
+	if dispName, ok := addressFields[DisplayNameFieldName].(*string); ok {
+		concatenatedAddress = fmt.Sprintf("%s,%s", concatenatedAddress, ptr.Val(dispName))
+	}
+
+	if address, ok := addressFields[AddressFieldName].(map[string]any); ok {
```

Comment:
naming a map a singular name (ex: address, field, etc) makes it seem like the value isn't a map, because maps contain multiple values.  
```suggestion
	if fields, ok := addressFields[AddressFieldName].(map[string]any); ok {
```